packages/salesforcedx-vscode-lwc/test/jest/testSupport/testRunner/testRunner.test.ts

```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright (c) 2023, salesforce.com, inc.
+ * All rights reserved.
+ * Licensed under the BSD 3-Clause license.
+ * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+
+import { getTestNamePatternArgs } from '../../../../src/testSupport/testRunner/testRunner';
+
+describe('testRunner Unit Tests.', () => {
+  describe('test getTestNamePatternArgs', () => {
+    it('Should return testNamePattern if flag is included', async () => {
+      const testName = 'Testing is Fun!';
+      const testPatternArgs = getTestNamePatternArgs(testName);
+      expect(testPatternArgs.length).toEqual(2);
+      expect(testPatternArgs).toContain('--testNamePattern');
+      expect(testPatternArgs).toContain(testName);
+      expect(testPatternArgs).toMatchSnapshot();
+    });
+    it('Should escape certain symbols if testNamePattern is included', async () => {
+      const testName = 'Test ?$^*().[]{}|+ Symbols';
+      const testNameEscaped = 'Test \\?\\$\\^\\*\\(\\)\\.\\[\\]\\{\\}\\|\\+ Symbols';
+      const testPatternArgs = getTestNamePatternArgs(testName);
+      expect(testPatternArgs).toContain(testNameEscaped);
+      expect(testPatternArgs).toMatchSnapshot();
+    });
+    it('Should not escape certain symbols if testNamePattern is included', async () => {
+      const testName = 'Test !@#"%&;:,<>\=~` Symbols';
+      const testPatternArgs = getTestNamePatternArgs(testName);
+      expect(testPatternArgs).toContain(testName);
+      expect(testPatternArgs).toMatchSnapshot();
```

Comment:
nit: So the toMatchSnapshot is checking the same thing as the toContain for all tests.

Reply:
Got it! I actually think I'm going to leave these in just since it's more readable for the explicit thing we're testing, and I like having the snapshot for the making sure it doesn't change

Reply:
cool.  It's not that big a deal.  Just don't burn it in as the way to do snapshot tests :).  One downside is if we change the regex to account for the exception cases we know about today the tests will all need to be touched. snapshot tests only would be more an an auto without code changes (jest would update the snaps).